projects/filjs-audit/README.md

```diff
@@ -0,0 +1,391 @@
+
+These projects should be placed into the GitHub "archived" state, with clear notices in their README about their status as not up-to-date examples. Further development work should not be ruled out on these projects, and a path to re-starting them should be documented in their README (e.g. "Contact ... at ... if you would like to re-start this project and contribute fixes to make it usable").
+
+### https://github.com/filecoin-shipyard/js-filecoin-api-client
+
+API client for go-filecoin.
+
+We should ask the maintainers of Venus if they would like to do anything with this in the short-term, if not, it should be archived.
+
+Last updated: 15 Sep 2019
+
+### https://github.com/filecoin-shipyard/filecoin-browse-asks
+
+> A simple command line tool to view the available Filecoin asks on the network.
+
+Last updated: 28 Sep 2019
+
+### https://github.com/filecoin-shipyard/filecoin-big-head
+
+> A simple command line tool to display the height of the Filecoin blockchain that your node has synced.
+
+Last updated: 28 Sep 2019
+
+### https://github.com/filecoin-shipyard/use-filecoin-network-info
+
+> React hook to query network info via Filecoin API
+
+Last updated: 28 Sep 2019
+
+### https://github.com/filecoin-shipyard/use-filecoin-head
+
+> React hook to query head blocks and chain height via Filecoin API
+
+Last updated: 28 Sep 2019
+
+### https://github.com/filecoin-shipyard/use-filecoin-config
+
+> React hook to load config via Filecoin API
+
+Last updated: 28 Sep 2019
+
+### https://github.com/filecoin-shipyard/filecoin-pickaxe
+
+> This is a command-line tool for managing "bundles" (groups of files) that you would like to import into Filecoin.
+
+Last updated: 29 Sep 2019
+
+#### https://github.com/filecoin-shipyard/filecoin-pickaxe-direct-deal
+
+> This is a command-line tool for interactively storing "bundles" of files from filecoin-pickaxe into the Filecoin network.
+
+Last updated: 28 Sep 2019
+
+#### https://github.com/filecoin-shipyard/filecoin-pickaxe-agent
+
+> Agent to perform requested Filecoin tasks
+
+Last updated: 7 Sep 2019
+
+### https://github.com/filecoin-shipyard/npm-go-filecoin-dep
+
+> Download go-filecoin to your node_modules.
+
+Last updated: 1 Sep 2019
+
+## Ecosystem libraries
+
+### Glif.io
+
+https://www.glif.io/
+
+#### https://github.com/glifio/wallet
+
+#### https://github.com/glifio/modules/tree/primary/packages
+
+Dependents of https://github.com/glifio/wallet, some are also useful by themselves.
+
+##### filecoin-address
+##### filecoin-message-confirmer
+##### filecoin-message
+##### filecoin-number
+
+Was https://github.com/glifio/filecoin-number
+
+##### filecoin-rpc-client
+##### filecoin-wallet-provider
+##### local-managed-provider
+##### react-components
```

Comment:
```suggestion
##### [@glif/filecoin-address](https://github.com/glifio/modules/tree/primary/packages/filecoin-address)

> This is a JS implementation of the Filecoin address type, inspired by go-address. It can create new address instances and encode addresses, and it takes care of decoding and validating checksums.

👍 use it. Adopt it into the filecoin-project org to give more visibilty.

##### [@glif/filecoin-message-confirmer](https://github.com/glifio/modules/tree/primary/packages/filecoin-message-confirmer)

Poll the StateSearchMsg API method to verify that a specfied message CID appears on-chain with a 0 exit-code reciept. Uses `@glif/filecoin-rpc-client`.

A reasonable solution given the current API. We should explore a better API for waiting for a message to succesfully execute.

Need to fix https://github.com/glifio/modules/issues/65

##### [@glif/filecoin-message](https://github.com/glifio/modules/tree/primary/packages/filecoin-message)

Normalises the casing of message object keys between lotus and zondax. If you need to talk to both, you are going to want to avoid dealing with the fact that lotus uses upper-camel case `GasFeeCap` and zondax uses all lower case `gasfeecap`.

##### [@glif/filecoin-number](https://github.com/glifio/modules/tree/primary/packages/filecoin-number)

A wrapper class for bignumber.js to represent an amount of FIL which may be larger than `Number.MAX_SAFE_INTEGER` (2^53 - 1). Provides conversion from attoFIL to picoFIL to FIL, and a Currency Conversion api which uses a remote service to provide the exchange rate.

Review and potentially replace with a lighter / more ergonimic alternative. 

- We should explore the ideal API for this representing FIL amounts.
- Consider using a lighter lib for big number support https://github.com/MikeMcl/big.js/wiki
- Consider just using the built-in BigInt https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt
- Currency conversion should be a seperate utility as many use-cases will not need it.

##### [@glif/filecoin-rpc-client](https://github.com/glifio/modules/tree/primary/packages/filecoin-rpc-client)

**TL;DR** Too bare-bones for general consumption, but a good example for folks who like it minimal.

A very light [(~70 LoC) wrapper](https://github.com/glifio/modules/blob/9b97f31054c20b39b8a75d30727fa33162715dab/packages/filecoin-rpc-client/src/index.ts
) around `axios` to create JSON RPC requests.

- 👍 lightweight, low maintenance, never out of date, easy to understand.
- 👍 lovely docs https://documenter.getpostman.com/view/4872192/SWLh5mUd?version=latest
- ❌ methods passed as string arg. No api, no types, no way to discover the api from your IDE.
- ❌ you've got to parse filecoin responses manually
- ❌ CID params are awkward. Must be passed as a ipld-json link `{ '/': '<cid string>' }`


##### [@glif/filecoin-wallet-provider](https://github.com/glifio/modules/tree/primary/packages/filecoin-wallet-provider)

Abstraction layer for filecoin wallet implementations. A good example to explore, but needs more iterations before it's something we should recommend to folks, and it's unlikely to get that attention in the near future.

Aims at being a facade over different filecoin impl providers, to provide a common api and re-use common / computable methods where possible. Ideally a povider impl adds implementations of sensitive functions / primatives while non-senstive / high-level ones are provided by this module.

##### [@glif/react-components](https://github.com/glifio/modules/tree/primary/packages/react-components)

Not for general consumption. Has some nice UI components, but folks can explore the wallet app to see them, and get the most up to date versions.
```